kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/storage/file/KieServerStateFileInit.java

```diff
@@ -0,0 +1,145 @@
+
+import java.io.File;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Properties;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.TreeSet;
+
+import org.kie.api.builder.ReleaseId;
+import org.kie.server.api.KieServerEnvironment;
+import org.kie.server.api.model.KieContainerResource;
+import org.kie.server.api.model.KieServerConfig;
+import org.kie.server.services.impl.release.ComparableReleaseId;
+import org.kie.server.services.impl.release.ReleaseIdComparator;
+import org.kie.server.services.impl.storage.KieServerState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Initialization class meant to be invoked by the OpenShift S2I process in bootstrapping an image with pre-installed KIE Containers.
+ */
+public class KieServerStateFileInit {
+
+    private static final Logger logger = LoggerFactory.getLogger(KieServerStateFileInit.class);
+
+    private KieServerStateFileInit() {}
+
+    public static void main(String... args) {
+        File serverStateFile = init();
+        logger.info("Initialized kie server state file: {}", serverStateFile);
+    }
+
+    public static File init() {
+        String serverRepo = getValue(KIE_SERVER_STATE_REPO, "KIE_SERVER_REPO", ".");
+        String serverId = getValue(KIE_SERVER_ID, "KIE_SERVER_ID", "kieserver");
+        KieServerEnvironment.setServerId(serverId);
+        KieServerEnvironment.setServerName(serverId);
+
+        File serverRepoDir = new File(serverRepo);
+        File serverStateFile = new File(serverRepoDir, serverId + ".xml");
+        if (serverStateFile.exists()) {
+            throw new IllegalStateException(String.format(
+                    "%s already exists. %s should only be used for pre-bootstrapping creation of server state file.",
+                    serverStateFile,
+                    KieServerStateFileInit.class.getSimpleName()));
+        }
+        KieServerStateFileRepository repository = new KieServerStateFileRepository(serverRepoDir);
+        KieServerState serverState = new KieServerState();
+
+        KieServerConfig config = new KieServerConfig();
+        Properties properties = new Properties();
+        properties.putAll(System.getProperties());
+        properties.put(KIE_SERVER_STATE_REPO, serverRepo);
+        properties.put(KIE_SERVER_ID, serverId);
+        repository.populateWithProperties(config, properties);
+        serverState.setConfiguration(config);
+
+        Set<KieContainerResource> containers = new LinkedHashSet<KieContainerResource>();
+        String serverContainerDeployment = getValue(KIE_SERVER_CONTAINER_DEPLOYMENT, "KIE_SERVER_CONTAINER_DEPLOYMENT", null);
+        if (serverContainerDeployment != null && !serverContainerDeployment.isEmpty()) {
+            Map<String,Set<ReleaseId>> containerIds_releaseIds = new TreeMap<String,Set<ReleaseId>>();
+            for (String unit : serverContainerDeployment.split("\\|")) {
```

Comment:
@sutaakar Sure, I can do it. I think that would help out @wmedvede's use of that format in other places. @mswiderski , sure I can document that. Thanks!